camayoc/tests/qpc/ui/test_credentials.py

```diff
@@ -0,0 +1,79 @@
+# coding=utf-8
+"""Tests for handling sources in the UI.
+
+:caseautomation: automated
+:casecomponent: ui
+:caseimportance: high
+:caselevel: integration
+:requirement: Sonar
+:testtype: functional
+:upstream: yes
+"""
+from uuid import uuid4
+
+from selenium.common.exceptions import NoSuchElementException
+
+from smartloc import Locator
+
+from widgetastic.exceptions import WidgetOperationFailed as wof
+from widgetastic.widget import Checkbox
+
+from widgetastic_patternfly import Button, Dropdown
+
+from .utils import clear_toasts, fill
+from .views import CredentialModalView, DashboardView, DeleteModalView
+
+
+def create_credential(view):
+    """Create a credential through the UI."""
+    dash = DashboardView(view)
+    dash.nav.select('Credentials')
+    # TODO: Parametrize for different types of credentials.
+    try:
+        add_credential_dropdown = Dropdown(view, 'Add Credential')
+        add_credential_dropdown.item_select('Network Credential')
+    except NoSuchElementException:
+        add_credential_dropdown = Dropdown(view, 'Add')
```

Comment:
So we have either "Add Credential" or "Add" for the same thing? Or is this just something that gets the text wrapped?

Reply:
This varies depending on whether or not a credential already exists. If there are no other credentials, it says "Add Credential" in the middle of the screen. If there is a credential on the screen already, it says "Add" in the top right corner of the screen.